Improve contributor docs around LLM use (#26358)
scala/scala3: LLM_POLICY.md

Reviewer: ```suggestion
- Do not contribute large volumes of LLM-generated code all at once. Try to scope them down to something digestible for code reviewers.

```

Reviewer: ```suggestion
- The code is well scoped, well documented and covered with tests.
```

The following rules are non-negotiable and necessary for a contribution to be considered.

- All interactions (issues, pull requests, discussions, etc.) must be 100% human-written.

>   Pasting LLM-generated text is not allowed, you must rewrite it in your own words to ensure you understand it.

Reviewer: ```suggestion
  LLM-generated text has to be processed and understood by a human, pasting LLM outputs verbatim is not allowed.
```
the policy shouldn't address the reader directly (so the word `you` is forbidden here 🙃)

- The author of the contribution is responsible for every line of code, comment, documentation, test or decision
  contained in it, regardless if it was done by hand or generated automatically by tooling.
- Any potential changes to requirements from a bug tracker issue must be discussed under the issue, not suggested in a PR.
- The author of a PR must be capable of answering reviewer questions and explaining their solution in detail.
- Contributed code should be covered by automated tests. If test automation is not feasible, an explanation has to be
  given in the pull request description, including a description of exactly how the change was tested manually.
  - Tests are not by themselves enough evidence that a contribution is good. We also care about maintainability,

>     and we acknowledge that our test coverage will never be 100% for edge cases so passing tests is not a guarantee of correctness.

Reviewer: ```suggestion
  - Tests are not enough evidence that a contribution is good all on their own. We also care about maintainability,
    and we acknowledge that our test coverage will never be 100% for edge cases, and so passing tests is not a guarantee of correctness.
```
